src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -124,4 +128,140 @@ public void createAccount_createsSingleValidAccount() throws Exception {
       assertThat(count).isEqualTo(1);
     }
   }
+
+  @Test
+  public void moveAccountBalance_validTransfer() throws Exception {
+    ByteArray fromAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    ByteArray toAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    BigDecimal fromAccountBalance = new BigDecimal(20);
+    BigDecimal toAccountBalance = new BigDecimal(0);
+    BigDecimal amount = new BigDecimal(10);
+    databaseClient.write(
+        ImmutableList.of(
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(fromAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(fromAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build(),
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(toAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(toAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build()));
+    spannerDao.moveAccountBalance(fromAccountId, toAccountId, amount);
+    try (ResultSet resultSet =
+        databaseClient
+            .singleUse()
+            .read(
+                "Account",
+                KeySet.newBuilder()
+                    .addKey(Key.of(fromAccountId))
+                    .addKey(Key.of(toAccountId))
+                    .build(),
+                Arrays.asList("AccountId", "Balance"))) {
+      int count = 0;
+      while (resultSet.next()) {
+        if (resultSet.getBytes(0).equals(fromAccountId)) {
+          assertThat(resultSet.getBigDecimal(1)).isEqualTo(fromAccountBalance.subtract(amount));
```

Comment:
[`getBigDecimalList`](https://googleapis.dev/java/google-cloud-spanner/6.6.1/com/google/cloud/spanner/StructReader.html#getBigDecimalList-int-) might make your life easier here.

Reply:
If I'm understanding how to use this function correctly, this gets the list of BigDecimals from a column, but we still need to retain the row relationship between accountId and balance and look for the specific entry.